### PR TITLE
feat: 🍰 Refine Group Creation And Group Edit

### LINKED ISSUE #5397: 🚀 [Feature] Refine Group Creation And Group Edit
## :rocket: Feature Request

Refine group creation and group edit.

### Todo

- [x] group icon on left side in header configurable (maybe three people)
- [ ] fields:
  - [x] general validation for all fields like on post
  - [ ] slug:
    - [ ] change does not work like expected
      - [ ] backend is involved
      - [ ] discuss with @Mogge and @ogerly how it should work
    - [x] created an issue #5480
  - [x] goal:
    - [x] add placeholder
  - [x] description:
    - [x] add placeholder
    - [x] add char counter like on post content
    - [x] tiptap editor
  - [x] locationName:
    - [x] have a clear button like in the search select in the header to empty the field
  - [x] Translation 
    - [x] translation input selects group radius and group type

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pull Request

- add tiptap editor in FormGroup.vue
- location Name wurd gelöscht
- validation of formSchema
- localisation of selects 

### Issues

- relates #5397 

 ### tip-tap editor
image

### clear location input
image

### localisation
https://user-images.githubusercontent.com/1324583/192789080-8ab6feda-f35c-42a3-919f-889e0277ff93.mp4

### CODE REVIEW

File: webapp/components/Group/GroupForm.vue [hunk 1]
```diff
@@ -161,6 +172,7 @@ export default {
     return {
       categoriesActive: this.$env.CATEGORIES_ACTIVE,
       disabled: false,
+      descriptionMin: 50,
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:175]: There is a constant for it in `webapp/constants/groups.js`.

Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:228]: donw

**Comment by ogerly** [comment]
FireShot Capture 1246 - ocelot social - ocelot social - localhost

FireShot Capture 1248 - ocelot social - ocelot social - localhost

File: webapp/components/Group/GroupForm.vue [hunk 2]
```diff
@@ -169,18 +224,18 @@ export default {
...
         categoryIds: categories ? categories.map((category) => category.id) : [],
       },
       formSchema: {
         name: { required: true, min: NAME_LENGTH_MIN, max: NAME_LENGTH_MAX },
-        slug: { required: false },
+        slug: { required: false, min: NAME_LENGTH_MIN },
         groupType: { required: true },
-        about: { required: true },
-        description: { required: true },
+        goal: { required: false, min: GOAL_LENGTH_MIN },
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:237]: There is no `min`, see above … 👆🏼

**Review by Tirokk — CHANGES_REQUESTED** [review]
I give a prereview and then go ahead with reviewing …

File: webapp/components/Group/GroupForm.vue [hunk 3]
```diff
@@ -7,133 +7,196 @@
...
+          <ds-space margin-top="large">
+            <ds-text class="select-label">
+              {{ $t('group.description') }}
+            </ds-text>
+            <hc-editor
+              :users="null"
+              :value="formData.description"
+              :hashtags="null"
+              @input="updateEditorDescription"
+            />
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue]: done

File: webapp/components/Group/GroupForm.vue [hunk 4]
```diff
@@ -142,6 +205,7 @@ export default {
   name: 'GroupForm',
   components: {
     CategoriesSelect,
+    HcEditor,
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:198]: done

File: webapp/components/Group/GroupForm.vue [hunk 5]
```diff
@@ -7,133 +7,196 @@
...
 import CategoriesSelect from '~/components/CategoriesSelect/CategoriesSelect'
 import { CATEGORIES_MIN, CATEGORIES_MAX } from '~/constants/categories.js'
-import { NAME_LENGTH_MIN, NAME_LENGTH_MAX } from '~/constants/groups.js'
+import {
+  NAME_LENGTH_MIN,
+  NAME_LENGTH_MAX,
+  GOAL_LENGTH_MIN,
+  DESCRIPTION_WITHOUT_HTML_LENGTH_MIN,
+} from '~/constants/groups.js'
+import HcEditor from '~/components/Editor/Editor'
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:189]: done

File: webapp/components/Group/GroupForm.vue [hunk 6]
```diff
@@ -7,133 +7,184 @@
...
-      <ds-input
-        v-if="update"
-        :label="$t('group.labelSlug')"
-        v-model="formData.slug"
-        icon="at"
-        :placeholder="`${$t('group.labelSlug')} …`"
-      ></ds-input>
-      <!-- groupType -->
-      <!-- TODO: change it has to be implemented later -->
-      <!-- TODO: move 'ds-select' from styleguide to main code and implement missen translation etc. functionality -->
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:24]: Would be nice do have this both todos in the code.
Or you don't think so?

File: webapp/components/Group/GroupForm.vue [hunk 7]
```diff
@@ -7,133 +7,184 @@
...
+          <!-- Group Slug -->
+          <ds-space margin-top="large">
+            <ds-input
+              v-if="update"
+              :label="$t('group.labelSlug')"
+              model="slug"
+              icon="at"
+              :placeholder="`${$t('group.labelSlug')} …`"
+            ></ds-input>
+          </ds-space>
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:36]: This does not work like expected if we edit in my eyes.
We have to talk about how it should wokr with @Mogge I guess and it has as well to do with the backend behaviour.

Will we do it in another separate issue?

File: webapp/components/Group/GroupForm.vue [hunk 8]
```diff
@@ -7,133 +7,184 @@
...
+            >
+              <option v-for="groupType in groupTypeOptions" :key="groupType" :value="groupType">
+                {{ $t(`group.types.${groupType}`) }}
+              </option>
+            </select>
+            <ds-text align="right">
+              <ds-chip size="base" :color="formData.groupType === '' ? 'danger' : 'medium'">
+                <base-icon v-if="formData.groupType === ''" name="warning" />
+              </ds-chip>
+            </ds-text>
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:60]: I would not show this chip if there is no error.
Or I would show a grey `1 / 1` if there is no error and not an empty chip.

Same on action radius …

File: webapp/components/Group/GroupForm.vue [hunk 9]
```diff
@@ -7,133 +7,184 @@
...
+                name="actionRadius"
+              >
+                {{ $t(`group.actionRadii.${actionRadius}`) }}
+              </option>
+            </select>
+            <ds-text align="right">
+              <ds-chip size="base" :color="formData.actionRadius === '' ? 'danger' : 'medium'">
+                <base-icon v-if="formData.actionRadius === ''" name="warning" />
+              </ds-chip>
+            </ds-text>
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:120]: See group type …

File: webapp/components/Group/GroupForm.vue [hunk 10]
```diff
@@ -156,11 +208,12 @@ export default {
     },
   },
   data() {
-    const { name, slug, groupType, about, description, actionRadius, locationName, categories } =
+    const { name, slug, groupType, goal, description, actionRadius, locationName, categories } =
       this.group
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:212]: `goal` is called `about` in the database. So this is not working …

File: webapp/components/Group/GroupForm.vue [hunk 11]
```diff
@@ -200,53 +253,30 @@ export default {
...
+    },
+    changeActionRadius(e) {
+      this.formData.actionRadius = e.target.value
+    },
+    updateEditorDescription(value) {
+      this.$refs.groupForm.update('description', value)
+    },
     submit() {
-      const { name, about, description, groupType, actionRadius, locationName, categoryIds } =
+      const { name, goal, description, groupType, actionRadius, locationName, categoryIds } =
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:271]: `goal` is called `about` in the database. So this is not working …

File: webapp/components/Group/GroupForm.vue [hunk 12]
```diff
@@ -200,53 +253,30 @@ export default {
...
         this.formData
       const variables = {
         name,
-        about,
+        goal,
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:275]: `goal` is called `about` in the database. So this is not working …

File: webapp/components/Group/GroupForm.vue [hunk 13]
```diff
@@ -169,18 +222,18 @@ export default {
         name: name || '',
         slug: slug || '',
         groupType: groupType || '',
-        about: about || '',
+        goal: goal || '',
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:225]: `goal` is called `about` in the database. So I would named it `about`…

File: webapp/components/Group/GroupForm.vue [hunk 14]
```diff
@@ -169,18 +222,18 @@ export default {
...
         categoryIds: categories ? categories.map((category) => category.id) : [],
       },
       formSchema: {
         name: { required: true, min: NAME_LENGTH_MIN, max: NAME_LENGTH_MAX },
-        slug: { required: false },
+        slug: { required: false, min: NAME_LENGTH_MIN },
         groupType: { required: true },
-        about: { required: true },
-        description: { required: true },
+        goal: { required: false },
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:235]: `goal` is called `about` in the database. So I would named it `about`…

File: webapp/components/Group/GroupForm.vue [hunk 15]
```diff
@@ -7,133 +7,184 @@
...
+                <base-icon v-if="formData.groupType === ''" name="warning" />
+              </ds-chip>
+            </ds-text>
+          </ds-space>
+
+          <!-- goal -->
+          <ds-space margin-top="large">
+            <ds-input
+              :label="$t('group.goal')"
+              v-model="formData.goal"
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:67]: `goal` is called `about` in the database. So I would named it `about`…

File: webapp/components/Group/GroupForm.vue [hunk 16]
```diff
@@ -7,133 +7,184 @@
...
+              :placeholder="$t('group.goal') + ' …'"
+              name="goal"
```
Comment (Tirokk) [inline webapp/components/Group/GroupForm.vue:69]: `goal` is called `about` in the database. So I would named it `about`…

**Review by Tirokk — CHANGES_REQUESTED** [review]
Cool that you have found out the validation problem with the selects. 😄

I would change some things …

The error message `name must be between 3 and 50 characters` is not translated.

Can we get rid of this grey strokes under the select if all is okay?
Or put `1 / 1` in the grey chip?

Bildschirmfoto 2022-10-02 um 12 48 29

We have different space between several things:

Bildschirmfoto 2022-10-02 um 12 56 29

File: webapp/constants/groups.js [hunk 17]
```diff
@@ -1,5 +1,6 @@
 // this file is duplicated in `backend/src/constants/group.js` and `webapp/constants/group.js`
 export const NAME_LENGTH_MIN = 3
 export const NAME_LENGTH_MAX = 50
+export const GOAL_LENGTH_MIN = 3
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/constants/groups.js]: is deleted

File: webapp/components/Group/GroupForm.vue (hunk 16 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:70]: is changed

File: webapp/components/Group/GroupForm.vue (hunk 15 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:68]: is changed

File: webapp/components/Group/GroupForm.vue (hunk 14 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:236]: is changed

File: webapp/components/Group/GroupForm.vue (hunk 13 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:226]: is changed

File: webapp/components/Group/GroupForm.vue (hunk 12 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:276]: is changed

File: webapp/components/Group/GroupForm.vue [hunk 18]
```diff
@@ -7,133 +7,184 @@
...
+              :placeholder="`${$t('group.labelSlug')} …`"
+            ></ds-input>
+          </ds-space>
+
+          <!-- groupType -->
+          <ds-space margin-top="large">
+            <ds-text class="select-label">
+              {{ $t('group.type') }}
+            </ds-text>
+            {{ formData.groupType }}
```
[the remark this answers is not in the document]
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:44]: deleted

File: webapp/components/Group/GroupForm.vue (hunk 6 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:24]: is now included

File: webapp/components/Group/GroupForm.vue (hunk 7 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue:37]: yes, let's make this a separate issue

Comment (Tirokk) [reply webapp/components/Group/GroupForm.vue:37]: Here it is already #5397

File: webapp/components/Group/GroupForm.vue (hunk 8 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue]: done

File: webapp/components/Group/GroupForm.vue (hunk 9 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue]: done

File: webapp/components/Group/GroupForm.vue (hunk 11 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue]: changed

File: webapp/components/Group/GroupForm.vue (hunk 10 above)
Comment (ogerly) [reply webapp/components/Group/GroupForm.vue]: changed

**Review by Tirokk — APPROVED** [review]
Very cool @ogerly ! 😍
Nice that you cared for the validations !!!  🚀🚀💫💫💫

I refined some things to make it even better, I hope.

**Comment by ogerly** [comment]
@Tirokk   i put another ds-space in the form when editing group data. there was no clean space between slug and groupType.

**Pull request merged**